sdk/trace/sampling.go

```diff
@@ -125,29 +125,101 @@ func NeverSample() Sampler {
 	return alwaysOffSampler{}
 }
 
-// ParentSample returns a Sampler that samples a trace only
-// if the the span has a parent span and it is sampled. If the span has
-// parent span but it is not sampled, neither will this span. If the span
-// does not have a parent the fallback Sampler is used to determine if the
-// span should be sampled.
-func ParentSample(fallback Sampler) Sampler {
-	return parentSampler{fallback}
+// ParentBased returns a composite sampler which behaves differently,
+// based on the parent of the span. If the span has no parent,
+// the root(Sampler) is used to make sampling decision. If the span has
+// a parent, depending on whether the parent is remote and whether it
+// is sampled, one of the following samplers will apply:
+// - remoteParentSampled(Sampler) (default: AlwaysOn)
+// - remoteParentNotSampled(Sampler) (default: AlwaysOff)
+// - localParentSampled(Sampler) (default: AlwaysOn)
+// - localParentNotSampled(Sampler) (default: AlwaysOff)
+func ParentBased(root Sampler, samplers ...SamplerOption) Sampler {
+	return parentBased{
+		root:     root,
+		samplers: configureSamplersForParentBased(samplers),
+	}
+}
+
+type parentBased struct {
+	root     Sampler
+	samplers parentBasedSamplers
+}
+
+type SamplerOption func(*parentBasedSamplers)
```

Comment:
Should this be `ParentBasedSamplerOption` since it appears to only apply to the `ParentBased` sampler?  Also, please add a docblock for this type.

Reply:
Didn't want to choose a too long of a name, but on second thought I agree, in this case it's probably better to be more specific.